### PR TITLE
Add Crust to dStorage section

### PR DESCRIPTION
**Pull request by badkk**

### CODE REVIEW

**Comment by bucanero** [comment]
Thanks for submitting this PR. We'll review the code snippets and ping back in case of any questions.

File: docs/concepts/decentralized-storage.md [hunk 1]
```diff
@@ -267,3 +268,231 @@ https://siasky.net/7ADpAwNWRxiIj_nMX3GfVbuGSJHRFdVMasopqq195Ua3Eg
 :::tip
 Please note that using Arweave's network will require purchasing artokens to pay for storage. Learn more at [arweave.org](https://www.arweave.org/).
 :::
+
+## Crust
+
+[Crust](https://crust.network) provides a Web3.0 decentralized storage network for the Metaverse. It is designed to realize core values of decentralization, privacy and assurance. Crust supports multiple storage-layer protocols such as IPFS, and exposes instant accessible on-chain storage functions to users. Crustʼs technical stack is also capable of supporting data manipulating and computing.
+
+The Crust protocol is 100% compatible with [IPFS](https://ipfs.io) procotol, it matches people who have hard drive space to spare with those users that need to store data or host content. Crust based on Polkadot ecosystem, but also supports the majority contract platforms, including Near/Solana/Ethereum/Elrond/... with its cross-chain solution.
```
Comment (bucanero) [inline docs/concepts/decentralized-storage.md:276]:
```suggestion
The Crust protocol is 100% compatible with [IPFS](https://ipfs.io) protocol, it matches people who have hard drive space to spare with those users that need to store data or host content. Crust is based on the Polkadot ecosystem, but also supports the majority of contract platforms, including Near/Solana/Ethereum/Elrond with its cross-chain solution.
```

File: docs/concepts/decentralized-storage.md [hunk 2]
```diff
@@ -267,3 +268,231 @@ https://siasky.net/7ADpAwNWRxiIj_nMX3GfVbuGSJHRFdVMasopqq195Ua3Eg
...
+        cid: cid.path,
+        size: fileStat.cumulativeSiz
+    };
+}
+```
+
+- With [IPFS W3Authed Gateway](https://docs.ipfs.io/concepts/ipfs-gateway/#authenticated-gateways)
+
+> You can find more `ipfsW3GW` endpoints on [LINK](https://github.com/crustio/ipfsscan/blob/main/lib/constans.ts#L29).
+> You can also find more `authHeader` web3 supports on [LINK](https://github.com/RoyTimes/crust-workshop/tree/master/src), the following exmaple just takes Near as example.
```
Comment (bucanero) [inline docs/concepts/decentralized-storage.md:318]:
```suggestion
> You can also find more `authHeader` web3 supports on [LINK](https://github.com/RoyTimes/crust-workshop/tree/master/src), the following sample just takes Near as an example.
```

**Comment by badkk** [comment]
> Thanks for submitting this PR. We'll review the code snippets and ping back in case of any questions.

Already solved the comments!

**Comment by badkk** [comment]
@bucanero Any feedback?

**Comment by think-in-universe** [comment]
thanks @badkk 

Crust is trying to support NEAR with seamless developer experience, so the tutorial would be helpful. 

My suggestion is that whether can move most the code snippets into an example project on GitHub, and keep the style of the Crust section consistent with the introduction to IPFS, Arweave and Sia.

File: docs/concepts/decentralized-storage.md [hunk 3]
```diff
@@ -267,3 +268,231 @@ https://siasky.net/7ADpAwNWRxiIj_nMX3GfVbuGSJHRFdVMasopqq195Ua3Eg
...
+:::info
+To learn more about Crust, check its [Decentralized Storage Market](https://wiki.crust.network/docs/en/DSM) and [Guaranteed Proof of Stake](https://wiki.crust.network/docs/en/GPoS)
+Also, you can start with [Build-101](https://wiki.crust.network/docs/en/build101).
+:::
+
+### Example of storing file with Crust and Near
+
+### 1. Upload file to IPFS
+
+> If you want to learn how to upload **FOLERS/FILES** into IPFS, please refer this [STEP](https://wiki.crust.network/docs/en/buildFileStoringWithGWDemo#1-upload-files-to-ipfs-gateway)
```
Comment (bucanero) [inline docs/concepts/decentralized-storage.md:287]:
```suggestion
> If you want to learn how to upload **FOLDERS/FILES** into IPFS, please refer to this [STEP](https://wiki.crust.network/docs/en/buildFileStoringWithGWDemo#1-upload-files-to-ipfs-gateway)
```

File: docs/concepts/decentralized-storage.md [hunk 4]
```diff
@@ -267,3 +268,231 @@ https://siasky.net/7ADpAwNWRxiIj_nMX3GfVbuGSJHRFdVMasopqq195Ua3Eg
 :::tip
 Please note that using Arweave's network will require purchasing artokens to pay for storage. Learn more at [arweave.org](https://www.arweave.org/).
 :::
+
+## Crust
+
+[Crust](https://crust.network) provides a Web3.0 decentralized storage network for the Metaverse. It is designed to realize core values of decentralization, privacy and assurance. Crust supports multiple storage-layer protocols such as IPFS, and exposes instant accessible on-chain storage functions to users. Crustʼs technical stack is also capable of supporting data manipulating and computing.
+
+The Crust protocol is 100% compatible with [IPFS](https://ipfs.io) protocol, it matches people who have hard drive space to spare with those users that need to store data or host content. Crust is based on the Polkadot ecosystem, but also supports the majority of contract platforms, including Near/Solana/Ethereum/Elrond/... with its cross-chain solution.
```
Comment (bucanero) [inline docs/concepts/decentralized-storage.md:276]:
```suggestion
The Crust protocol is 100% compatible with [IPFS](https://ipfs.io) protocol, it matches people who have hard drive space to spare with those users that need to store data or host content. Crust is based on the Polkadot ecosystem, but also supports the majority of contract platforms, including NEAR/Solana/Ethereum/Elrond/etc. with its cross-chain solution.
```

File: docs/concepts/decentralized-storage.md [hunk 5]
```diff
@@ -267,3 +268,231 @@ https://siasky.net/7ADpAwNWRxiIj_nMX3GfVbuGSJHRFdVMasopqq195Ua3Eg
...
+Also, you can start with [Build-101](https://wiki.crust.network/docs/en/build101).
+:::
+
+### Example of storing file with Crust and Near
+
+### 1. Upload file to IPFS
+
+> If you want to learn how to upload **FOLERS/FILES** into IPFS, please refer this [STEP](https://wiki.crust.network/docs/en/buildFileStoringWithGWDemo#1-upload-files-to-ipfs-gateway)
+
+First, you need to put your file into IPFS, there's 2 ways to upload file to IPFS: local IPFS node or remote IPFS gateway:
```
Comment (bucanero) [inline docs/concepts/decentralized-storage.md:289]:
```suggestion
First, you need to put your file into IPFS. There are 2 ways to upload files to IPFS: local IPFS node or remote IPFS gateway:
```

**Comment by badkk** [comment]
> thanks @badkk
> 
> Crust is trying to support NEAR with seamless developer experience, so the tutorial would be helpful.
> 
> My suggestion is that whether can move most the code snippets into an example project on GitHub, and keep the style of the Crust section consistent with the introduction to IPFS, Arweave and Sia.

Sure! already resolved!

**Comment by badkk** [comment]
> > @bucanero Any feedback?
> 
> Hello @badkk , sorry for the delay, we had some busy weeks due to the ETH Denver conference.
> 
> As @think-in-universe proposed, I'd suggest creating a public GitHub repo with the implementation example for NEAR, so anyone can check it out and test it. You can keep the example on the article, but there's no need to add the code on the document, as it will be available on the example GitHub repo.
> 
> Thanks!

Oh, np! I've already put all code into the demo repo 😀

**Pull request merged**